Firestore/core/test/firebase/firestore/local/leveldb_key_test.cc

```diff
@@ -356,6 +371,59 @@ TEST(RemoteDocumentKeyTest, Description) {
       LevelDbRemoteDocumentKey::Key(testutil::Key("foo/bar/baz/quux")));
 }
 
+TEST(RemoteDocumentReadTimeKeyTest, Ordering) {
+  // Different collection paths:
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("bar", 1),
+            RemoteDocumentReadTimeKeyPrefix("baz", 1));
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("bar", 1),
+            RemoteDocumentReadTimeKeyPrefix("foo/doc/bar", 1));
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("foo/doc/bar", 1),
+            RemoteDocumentReadTimeKeyPrefix("foo/doc/baz", 1));
+
+  // Different read times:
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("foo", 1),
+            RemoteDocumentReadTimeKeyPrefix("foo", 2));
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("foo", 1),
+            RemoteDocumentReadTimeKeyPrefix("foo", 1000000));
+  ASSERT_LT(RemoteDocumentReadTimeKeyPrefix("foo", 1000000),
+            RemoteDocumentReadTimeKeyPrefix("foo", 1000001));
+
+  // Different document ids:
+  ASSERT_LT(RemoteDocumentReadTimeKey("foo", 1, "a"),
+            RemoteDocumentReadTimeKey("foo", 1, "b"));
+}
+
+TEST(RemoteDocumentReadTimeKeyTest, EncodeDecodeCycle) {
+  LevelDbRemoteDocumentReadTimeKey key;
+
+  std::vector<std::string> collection_paths{"foo", "foo/doc/bar",
+                                            "foo/doc/bar/doc/baz"};
+  std::vector<int64_t> versions{1, 1000000, 1000001};
+  std::vector<std::string> document_ids{"docA", "docB"};
+
+  for (auto&& collection_path : collection_paths) {
```

Comment:
`auto&&` is "full magic", but we don't really need that here. `const auto&` will work just as well here and more clearly communicates that we're not mutating the values.

Reply:
Fixed.